internal/fsutil/walk_test.go

```diff
@@ -10,139 +10,170 @@ import (
+					return errSentinel
 				}
-				content = string(bs)
-				return sentinelErr
 			}
-		}
-		return nil
+
+			errOurs := fsutil.Walk(root, createWalkFn(&ourFiles))
+			Expect(errOurs).To(MatchError(errSentinel))
+
+			errTheirs := filepath.Walk(root, createWalkFn(&theirFiles))
+			Expect(errTheirs).To(MatchError(errSentinel))
+
+			sortBFSOrder(theirFiles)
+
+			Expect(ourFiles).To(Equal(theirFiles))
+		})
+
 	})
-	if !errors.Is(err, sentinelErr) {
-		t.Errorf("incorrect return value, expected sentinel error but got: %v", err)
-	}
-	if content != "a1/a2/a.txt" {
-		t.Errorf("unexpected content: %q", content)
-	}
-}
 
-func TestWalkSearchWithSkipDir(t *testing.T) {
-	root := createTestDir(t)
-	sentinelErr := errors.New("sentinel")
-	var content string
-	err := fsutil.Walk(root, func(path string, fi fs.FileInfo, err error) error {
-		if err != nil {
+	it("finds a file", func() {
+		var content string
+		err := fsutil.Walk(root, func(path string, fi fs.FileInfo, err error) error {
+			if err != nil {
+				return nil
+			}
+			if fi.Mode().Type() == 0 {
+				_, name := filepath.Split(path)
+				if name == filename {
+					bs, e := os.ReadFile(path)
+					if e != nil {
+						return e
+					}
+					content = string(bs)
+					return errSentinel
+				}
+			}
 			return nil
-		}
-		rp, err := filepath.Rel(root, path)
-		if rp == "a1" { // skip the first dir containing a.txt
-			return filepath.SkipDir
-		}
-		if fi.Mode().Type() == 0 {
-			_, name := filepath.Split(path)
-			if name == "a.txt" {
-				bs, e := os.ReadFile(path)
-				if e != nil {
-					return e
+		})
+
+		Expect(err).To(MatchError(errSentinel))
+		Expect(content).To(Equal(contentA1A2))
+	})
+
+	it("finds a file (skipping a dir)", func() {
+		var content string
+		err := fsutil.Walk(root, func(path string, fi fs.FileInfo, err error) error {
+			if err != nil {
+				return nil
+			}
+			rp, err := filepath.Rel(root, path)
+			Expect(err).NotTo(HaveOccurred())
+			if rp == "a1" { // skip the first dir containing a.txt
+				return filepath.SkipDir
+			}
+			if fi.Mode().Type() == 0 {
+				_, name := filepath.Split(path)
+				if name == filename {
+					bs, e := os.ReadFile(path)
+					if e != nil {
+						return e
+					}
+					content = string(bs)
+					return errSentinel
 				}
-				content = string(bs)
-				return sentinelErr
 			}
-		}
-		return nil
+			return nil
+		})
+
+		Expect(err).To(MatchError(errSentinel))
+		Expect(content).To(Equal(contentC1A2A3))
 	})
-	if !errors.Is(err, sentinelErr) {
-		t.Errorf("incorrect return value, expected sentinel error but got: %v", err)
-	}
-	if content != "c1/a2/a3/a.txt" {
-		t.Errorf("unexpected content: %q", content)
-	}
+}
+
+func sortBFSOrder(files []fileInfo) {
+	sort.SliceStable(files, func(i, j int) bool {
+		iLen := len(strings.Split(files[i].Path, string(filepath.Separator)))
+		jLen := len(strings.Split(files[j].Path, string(filepath.Separator)))
+		return iLen < jLen
+	})
+
 }
 
 func createTestDir(t *testing.T) string {
```

Comment:
```suggestion
// Create this file structure for the tests to run (obtained using `tree`)
// ├── a.lnk -> a1
// ├── a1
// │   ├── a2
// │   │   ├── a.txt
// │   │   ├── a3
// │   │   ├── b3
// │   │   └── c3
// │   ├── b2
// │   │   ├── a3
// │   │   ├── b3
// │   │   └── c3
// │   └── c2
// │       ├── a3
// │       ├── b3
// │       └── c3
// ├── b1
// │   ├── a2
// │   │   ├── a3
// │   │   ├── b3
// │   │   └── c3
// │   ├── b2
// │   │   ├── a3
// │   │   ├── b3
// │   │   └── c3
// │   └── c2
// │       ├── a3
// │       ├── b3
// │       └── c3
// └── c1
//
//	├── a2
//	│   ├── a3
//	│   │   └── a.txt
//	│   ├── b3
//	│   └── c3
//	├── b2
//	│   ├── a3
//	│   ├── b3
//	│   └── c3
//	└── c2
//	    ├── a3
//	    ├── b3
//	    └── c3
func createTestDir(t *testing.T) string {
```

Reply:
I wanted to add this when writing original tests but IDE kept reformatting it.